### PR TITLE
Minor fixes for distributions

### PR DESCRIPTION
**Pull request by mppf**
These fixes solve compilation problems that occur on my arrays branch. The arrays branch instantiates chpl__initCopy for each array type and that seems to reveal some simple problems with type mismatches.

The main non-trivial change here is to adjust BlockArr.dsiSupportsBulkTransfer and BlockArr.dsiSupportsBulkTransferInterface to return false when a sparseLayoutType has been set to something other than DefaultDist. This avoids compilation errors on the arrays branch when such an array's initCopy is resolved (and the initCopy in turn tries to use dsiSlice which is probably wrong - see the comment added there).

Note that sparse arrays don't currently support slicing at all.

Passed quickstart testing, local testing, and GASNet testing of distributions/ and release/examples/.
Reviewed by @benharsh - thanks!

### CODE REVIEW

**Comment by mppf** [comment]
- [ ] full local testing
- [ ] GASNet testing of distributions and release/examples

**Comment by mppf** [comment]
@benharsh - could you look at this?

File: modules/dists/CyclicDist.chpl [hunk 1]
```diff
@@ -885,7 +885,12 @@ proc CyclicArr.dsiPrivatize(privatizeData) {
 }
 
 
-inline proc _remoteAccessData.getDataIndex(param stridable, myStr: rank*idxType, ind: rank*idxType, startIdx, dimLen) {
+inline proc _remoteAccessData.getDataIndex(
+    param stridable, myStr:
+    rank*chpl__signedType(idxType),
```
Comment (benharsh) [inline modules/dists/CyclicDist.chpl]: Formatting mixup here: I think `myStr` should be on the next line.

**Comment by benharsh** [comment]
The rest looks reasonable to me 👍

**Pull request merged**